client/client.go

```diff
@@ -79,36 +83,70 @@ type Client struct {
 
 	warningCount     int
 	warningTimestamp time.Time
+
+	getWebsocket getWebsocketFunc
+}
+
+type getWebsocketFunc func(url string) (clientWebsocket, error)
+
+type clientWebsocket interface {
+	wsutil.MessageReader
+	wsutil.MessageWriter
+	io.Closer
+	jsonWriter
+}
+
+type jsonWriter interface {
+	WriteJSON(v interface{}) error
 }
 
 // New returns a new instance of Client
-func New(config *Config) *Client {
+func New(config *Config) (*Client, error) {
```

Comment:
The change looks good, and okay to do right now. But soon we won't be able to tolerate that kind of change in the client package, as this is exactly the part of Pebble that is supposed to be used as a library by third-parties. The exposed API needs to be rock-solid.